docs/architecture/grpc-for-wcf-developers/application-performance-management.md

```diff
@@ -0,0 +1,227 @@
+
+Because DiagnosticSource is a part of the core framework, it's supported by several core components, including <xref:System.Net.Http.HttpClient>, Entity Framework Core and ASP.NET Core, including explicit support in the gRPC framework. When ASP.NET Core receives a request, it  checks for a pair of HTTP headers matching the [W3C Trace Context](https://www.w3.org/TR/trace-context) standard. If the headers are found, an Activity is started using the identity values and context from the headers. If no headers are found, an Activity is started with generated identity values that match the standard format. Any diagnostics generated by the framework or by application code during the lifetime of this Activity can be tagged with the trace and span identifiers. The `HttpClient` support extends this further by checking for a current Activity on every request and automatically adding the trace headers to the outgoing request.
+
+The ASP.NET Core gRPC client and server libraries include explicit support for DiagnosticSource and Activity, and will create activities and apply and use header information automatically.
+
+> [!NOTE]
+> All of this only happens if a *listener* is consuming the diagnostic information. If there's no listener, no diagnostics are written and no activities are created.
+
+### Implementing your own DiagnosticSource
+
+Although a good quantity of data is automatically generated by ASP.NET Core, including gRPC, as well as Entity Framework Core and `HttpClient`, you may wish to add your own diagnostics or create explicit spans within your application code. You can create a named `DiagnosticSource` and write event data to it, or create an `Activity` to record a specific operation. `DiagnosticSource` is an abstract class; `DiagnosticListener` inherits from it and can be used to create an instance. The class is intended to be used as a singleton; it's common to create it as a static field in an internal class.
+
+```csharp
+internal static class StockDataDiagnostics
+{
+    public static readonly DiagnosticListener Listener = new DiagnosticListener("StockData");
+}
+```
+
+#### Writing an event to a DiagnosticListener
+
+To avoid the overhead of writing diagnostics when no consumer is listening, you should always call the `IsEnabled` method on the listener before writing to it. If a consumer is listening, you can write an event providing a name and an object with information about the event. This can be any object: a gRPC request, a type from your own code, or even an anonymous object.
```

Comment:
Would it be better to point to the already quite extensive examples and documentation for DiagnosticSource/Listener? This seems like it's redundant content as there's nothing gRpc specific here.